model/Check/System/CertificateCheck.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoSystemStatus\model\Check\System;
+
+use common_report_Report as Report;
+use oat\taoSystemStatus\model\Check\AbstractCheck;
+use oat\oatbox\log\loggerawaretrait;
+use DateTime;
+
+/**
+ * Class CertificateCheck
+ * @package oat\taoSystemStatus\model\Check\System
+ * @author Aleksej Tikhanovich, <aleksej@taotesting.com>
+ */
+class CertificateCheck extends AbstractCheck
+{
+    use LoggerAwareTrait;
+
+    /**
+     * @inheritdoc
+     */
+    protected function doCheck(): Report
+    {
+        $certInfo = $this->getCertInfo();
+
+        $now = new DateTime();
+        $validTo = new DateTime('@' . $certInfo['validTo_time_t']);;
+        $diffDays = $validTo->diff($now)->format("%a");
+        $data = $validTo->format('Y-m-d H:i:s');
+
+        if ($diffDays < 7) {
```

Comment:
Such values make sense to move to the class constants. In the future it easier to adjust them without producing diff in the implementation.